data/lib/compat/compat.lua

```diff
@@ -1006,3 +1006,13 @@ end
 function Guild.removeMember(self, player)
 	return player:getGuild() == self and player:setGuild(nil)
 end
+
+function doTileAddItemEx(pos, uid, flags)
+	flags = flags or 0
+	local tile = Tile(pos)
+	local item = Item(uid)
+	if tile and item then
+		tile:addItemEx(item, flags)
+	end
+	return false
```

Comment:
It's far more common to invert the conditions and fail fast. You'd usually do:

```lua
local tile = Tile(pos)
if not tile then
    return false
end

local item = Item(uid)
if not item then
    return false
end

-- do what happens when everything is ok
return true
```

And you don't need to check if `item` is not null because you already check on C++ (`getUserdata` then checking if it's not null).